### ISSUE TITLE
Game executable crashes since new patch (13/09/2022)

### ISSUE THREAD

**Issue by strigefleur** [Healix/Gw2Launcher#206]
Game patched via launcher-integrated updater.
Standalone run (without launcher) causes no problems.
Launcher run crashes the game the moment it tries to display login window.

**Comment by jbran** [comment]
Also seeing this since the 2022-09-13 patch: Part of error:

*--> Crash <--*
Assertion: Launcher: Browser crashed 
File: D:\Perforce\Live\NAEU\v2\Code\Gw2\Game\Launcher\LauncherCoherent.cpp(2559)
App: Gw2-64.exe 
Pid: 15432
BaseAddr: 00007FF758E00000
ProgramId: 101
Build: 134163
When: 2022-09-13T16:23:51Z 2022-09-13T09:23:51-07:00
Uptime:   0 days  0:00:01
Flags: 0
DumpFile: Crash.dmp




(as an aside, does Anet use perforce?)

**Comment by zosek08** [comment]
Same issue over here. Was using the latest beta release.

**Comment by Lorkster** [comment]
Same issue, running latest beta. Can provide crash logs if needed.

**Comment by Healix** [comment]
After all these years, CoherentUI no longer requires exclusive access and will instead crash if it can't use the default path; previously it would use an alternative location.

These options are no longer supported and will cause this crash if enabled:

Settings > Guild Wars 2 > Management > 
Localized CoherentUI: Give each account its own instance of CoherentUI. 
Localized execution set to binaries.

**Comment by jbran** [comment]
Thank you Healix, that did the trick!

**Comment by glaville** [comment]
Even without Localized CoherentUI and Localized execution, I encounter the "Launcher: Browser crashed" error as well:
are there other options to change to fix the problem?

![image](https://user-images.githubusercontent.com/321066/189993811-9814bb43-6f69-4e4d-92e1-ba6842702d24.png)

**Comment by Lorkster** [comment]
I can launch fine with default launcher, but disabling the two Localized did nothing (they weren't enabled). My wife is running the old release version and that works fine for her.

**Comment by Lorkster** [comment]
> enabling localized execution set to full

That worked for me! Thanks!

**Comment by glaville** [comment]
Yes, GW2 works fine when launched directly by clicking on the Gw2-64 binary.

I tried deleting the whole bin64 folder and letting GW2 recreate it, and repairing the game but it still crashes with the following Error:

```
*--> Error Logs <--*
20:02:04Z Coherent Error: PID: 18188 | 22252 20:02:04.089171 [18188:22252:FATAL:content_main_runner.cc(742)] Check failed: base::i18n::InitializeICU(). 
20:02:04Z Coherent Error: PID: 18188 | 22252 20:02:04.979368 [18188:22252:FATAL:(0)] Check failed: base::i18n::InitializeICU(). 
20:02:05Z Coherent Error: PID: 18188 | 296 20:02:05.000373 [5048:16544:FATAL:content_main_runner.cc(742)] Check failed: base::i18n::InitializeICU(). 
20:02:05Z Browser view crashed
Crashed at 20:02:05Z
```

If I set the option "Enable localized execution" to "full", the crash no longer occurs.

**Comment by Healix** [comment]
> I tried deleting the whole bin64 folder and letting GW2 recreate it and repairing the game but it still crashes with the following 

That is the same crash, which happens when CoherentUI can't access bin64/icudtl.dat. The options I previously mentioned are the only options that would cause that. Restart Gw2Launcher, confirm the settings are disabled, then try launching. Alternatively...

The current [beta version](https://github.com/Healix/Gw2Launcher/raw/master/Gw2Launcher/bin64/Beta/Gw2Launcher.exe) should fix the problem and the options I previously mentioned can be used again - turns out GW2 still supports it, they're just no longer handling it from CoherentUI's side. If it's _still_ a problem, something else is locking icudtl.dat and you'll need to enable "Localized CoherentUI" or "Localized execution" set to full.

**Comment by glaville** [comment]
Thanks, after disabling localized execution, fully closing Gw2launcher, and relaunching it, it now works against: the rogue lock on the file appears to have disappeared!

**Comment by 1Nor1** [comment]
thanks this worked out but ive recently keep encountering eternally stuck on loading screen (didnt happen before)fully updated nvidia gtx card, using beta version and after a while of leaving it on loading screen i get error about couldnt connect to server...

**Comment by maxz-GitHub** [comment]
Had the same issue.
Unticking "own instance of CoherentUI" and changing Desired Level to "Full" made it work again.

You are a champ Mr Healix :)

**Comment by Healix** [comment]
> thanks this worked out but ive recently keep encountering eternally stuck on loading screen (didnt happen before)fully updated nvidia gtx card, using beta version and after a while of leaving it on loading screen i get error about couldnt connect to server...

Getting stuck loading a map is typically caused by internet problems / packet loss (applies to any map), a blocked IP (only applies to maps on the affected IPs) or a lockout on the character (only applies to the affected character). Use -maploadinfo if you want a little more information or to make it more obvious when it's stuck.

The problem will likely fix itself within a day (assuming there's isn't a major problem like when SEA got its undersea cable damaged) or with the next patch (GW2 changes the IPs it uses with every patch).

**Comment by EukerionGW2** [comment]
Hi all,

Chipping in to say that disabling "Localized CoherentUI: Give each account its own instance of CoherentUI" and setting "Localized execution" to "Full" did the trick for me. The crashes no longer happen.

However I've noticed since last GW2 patch that each instance of the game (each account I launch) now has its own, separate icon in the taskbar, even though the "Prevent taskbar grouping" option is disabled. Basically I haven't changed any settings in a few weeks, I haven't updated GW2Launcher, and this suddenly started happening when the new GW2 patch was released (it was already there before I changed the "Localized execution" settings as stated before). Any ideas what's happening here? :)

**Comment by Healix** [comment]
> However I've noticed since last GW2 patch that each instance of the game (each account I launch) now has its own, separate icon in the taskbar, even though the "Prevent taskbar grouping" option is disabled. Basically I haven't changed any settings in a few weeks, I haven't updated GW2Launcher, and this suddenly started happening when the new GW2 patch was released (it was already there before I changed the "Localized execution" settings as stated before). Any ideas what's happening here? :)

Are you sure it stopped grouping without changing anything? Localized execution set to full will also prevent grouping, since Windows groups programs based on the path they were started from.

Note that localized execution isn't needed to fix the crashing problem, it's just an alternative. Disabling "give each account its own instance of CoherentUI" and restarting Gw2Launcher should fix it, or you can just use the latest [beta version](https://github.com/Healix/Gw2Launcher/raw/master/Gw2Launcher/bin64/Beta/Gw2Launcher.exe).

If you want to use grouping with localized execution set to full, the beta version has a few extra in-progress settings. Launch the beta version once to create the settings file, then exit (note these settings are only loaded on launch). Open the following file in notepad and change the following line:
%appdata%\Gw2Launcher\data\temp\settings.txt
ForceTaskbarGrouping=1

**Comment by wharlox** [comment]
Wanted to share that I'm using build 66511 (64-bit), Release 16, Date 2021-09-24 and I am not getting crashes as stated above; though my Settings -> Guild Wars 2 -> Management only has Mode: Virtual and other checkboxes un-checked.

Have one game local.dat and others are marked as shared with the launcher feeding in the creds via the copy-pasta method.

Only issues I have are the bait-n-switch throttling I get when trying to login to GW2 since I have a handful of accounts and gw2 does not let me login too rapidly due to them wanting my money but refuse proper service.
